src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +956,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+
+                List<string> primaryKeyColumnNames = new();
+                foreach (string primaryKey in currentEntitySourceDefinition.PrimaryKey)
+                {
+                    if (sqlMetadataProvider.TryGetExposedColumnName(entityName, primaryKey, out string? exposedPrimaryKeyName) && !string.IsNullOrWhiteSpace(exposedPrimaryKeyName))
+                    {
+                        primaryKeyColumnNames.Add(exposedPrimaryKeyName);
+                    }
+                }
+
+                dbResultSetForCurrentEntity = queryExecutor.ExecuteQuery(queryString,
+                                                                         queryParameters,
+                                                                         queryExecutor.ExtractResultSetFromDbDataReader,
+                                                                         GetHttpContext(),
+                                                                         primaryKeyColumnNames,
+                                                                         dataSourceName);
+
+                dbResultSetRowForCurrentEntity = dbResultSetForCurrentEntity is not null ? (dbResultSetForCurrentEntity.Rows.FirstOrDefault() ?? new DbResultSetRow()) : null;
+
+                if (dbResultSetRowForCurrentEntity is not null && dbResultSetRowForCurrentEntity.Columns.Count == 0)
+                {
+                    // For GraphQL, insert operation corresponds to Create action.
+                    throw new DataApiBuilderException(message: $"Could not insert row with given values for entity: {entityName}",
+                                                      statusCode: HttpStatusCode.Forbidden,
+                                                      subStatusCode: DataApiBuilderException.SubStatusCodes.DatabasePolicyFailure);
+                }
+
+                if (dbResultSetRowForCurrentEntity is null)
+                {
+                    throw new DataApiBuilderException(
+                        message: "No data returned back from database.",
+                        statusCode: HttpStatusCode.InternalServerError,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.DatabaseOperationFailed);
+                }
+
+                Dictionary<string, object?> insertedValues = dbResultSetRowForCurrentEntity.Columns;
+                Dictionary<string, object?> pkValues = new();
+                foreach (string pk in primaryKeyColumnNames)
+                {
+                    pkValues.Add(pk, insertedValues[pk]);
+                }
+
+                primaryKeysOfCreatedItem.Add(entityName, pkValues);
+                multipleCreateStructure.CurrentEntityPKs = pkValues;
+
+                //Perform an insertion in the linking table if required
+                if (multipleCreateStructure.IsLinkingTableInsertionRequired)
+                {
+                    if (multipleCreateStructure.LinkingTableParams is null)
```

Comment:
In the constructor for `MultipleCreateStructure`, this is always intialized when the `isLinkingTableInsertionRequired` is set to true. 

On that note, do we need to store `LinkingTableParams`/`isLinkingTableInsertionRequired` in the MultipleCreateStructure? Can't we directly determine during the recursion? The only condition this relies on is - the relationship should be an M:N relationship.

Reply:
We need `LinkingTableParams` definitely. When constructing the fields for the current entity, there's opportunity to identify and store scalar fields belonging to current and linking entity. Later we just populate the referencing fields for both current and linking entity. 

Storing the linking entity fields earlier lets us avoid the re-computation of identifying the fields belonging to linking table. 


I'll take another look at whether we need `isLinkingTableInsertionRequired`

Reply:
Having the field `isLinkingTableInsertionRequired` in `MultipleCreateStrucutre` makes the flow more intuitive. 

Consider the example of Book and Author entities in our test schema. There exists a M:N relationship b/w these two entities. 

In the current implementation, when the recursion flow is processing `Book` entity, whether the linking table insertion will be required along with creating `Author` is determined. When the flow is at `Book` entity, there is access to the relationship field name b/w `Book` and `Author` which is needed for determining whether the relationship type is `M:N`.  

If we were to not have this field, we will be evaluating this when the flow is processing `Author` entity at which point we don't have access to the relationship field name b/w `Author` and `Book` (parent level entity). To overcome this, we need to pass the relationship name or perform this evaluation and pass this is as a parameter in the recursive function. Relationship name is not persisted anywhere, the only place it is used is in `SourceDefinition.SourceEntityRelationshipMap`, where the relationship name is the key. So, writing a logic to determine whether the relatiionship type is M:N based on two entity names will not be O(1) operation (while evaluating the same using  <entity name, relationship name> is O(1)). 

Having the `isLinkingTableInsertionRequired` field is the better and cleaner approach.